plugins/bigquery/dbt/adapters/bigquery/impl.py

```diff
@@ -745,6 +746,11 @@ def get_table_options(
             expiration = 'TIMESTAMP_ADD(CURRENT_TIMESTAMP(), INTERVAL 12 hour)'
             opts['expiration_timestamp'] = expiration
 
+        if (config.get('time_to_expiration') is not None) and (not temporary):
+            expiration = ('TIMESTAMP_ADD(CURRENT_TIMESTAMP(), INTERVAL '
+            '{} hour').format(config.get('time_to_expiration'))
```

Comment:
The name 'time_to_expiration' doesn't provide any hints about the unit.  Maybe this could be 'hours_to_expiration'?

Reply:
@jtcohen6 How do you feel about this, following #2697?

Reply:
I'm totally in favor of `hours_to_expiration`